hpraid_exporter.go

```diff
@@ -170,17 +177,45 @@ func newHpraidExporter(utilityPath string, temporaryZipPath string) (*HpraidExpo
 	}, nil
 }
 
+func initTemporaryZipPath() {
+	var err error
+	temporaryZipPath, err = ioutil.TempDir("", "hpraid")
+	if err != nil {
+		log.Fatalf("failed to create temporary zip path: %s", err)
+	}
+	log.Debug("Using %s as temporary zip directory", temporaryZipPath)
+}
+
+func setupExitHandler() {
+	// Properly clean up before exit'ing on kill or Ctrl+C
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		<-sigs
+		log.Info("Cleaning up")
+		cleanTemporaryZipPath()
+		os.Exit(0)
+	}()
+}
+
+func cleanTemporaryZipPath() {
+	os.RemoveAll(temporaryZipPath)
+}
+
 func main() {
 	var (
-		listenAddress    = flag.String("web.listen-address", ":9423", "Address to listen on for web interface and telemetry.")
-		metricsPath      = flag.String("web.telemetry-path", "/metrics", "Path under which to expose metrics.")
-		utilityPath      = flag.String("hpraid.utility-path", "ssacli", "Path of the ssacli utility.")
-		temporaryZipPath = flag.String("hpraid.temporary-zip-path", "/tmp/hpraid_exporter.zip", "Path where a temporary zip file may be stored.")
+		listenAddress = flag.String("web.listen-address", ":9423", "Address to listen on for web interface and telemetry.")
+		metricsPath   = flag.String("web.telemetry-path", "/metrics", "Path under which to expose metrics.")
+		utilityPath   = flag.String("hpraid.utility-path", "ssacli", "Path of the ssacli utility.")
 	)
 	flag.Parse()
 
 	log.Info("Starting hpraid_exporter")
-	exporter, err := newHpraidExporter(*utilityPath, *temporaryZipPath)
+
+	initTemporaryZipPath()
```

Comment:
Instead of putting this here, could you please move this into `Collect()` or `collectFromUtility()`? As in, that we create a new temporary directory during every iteration. This is a lot more robust than doing it once. If someone decides to clean up `/tmp` or accidentally change some permissions, it may break the exporter until restarted.

Reply:
Thanks for the feedback. I get your point, but I would have liked it even more if you gave that hint in advance (as I specifically pointed out my plan to implement it during startup in #1). ;)
Integrated the improvements as requested nevertheless.